WebHostLib/options.py

```diff
@@ -173,16 +173,23 @@ def generate_yaml(game: str):
             else:
                 options[key] = val
 
-        # Detect and build ItemDict options from their name pattern
         for key, val in options.copy().items():
             key_parts = key.rsplit("||", 2)
+            # Detect and build ItemDict options from their name pattern
             if key_parts[-1] == "qty":
                 if key_parts[0] not in options:
                     options[key_parts[0]] = {}
                 if val != "0":
                     options[key_parts[0]][key_parts[1]] = int(val)
                 del options[key]
 
+            # Detect keys which end with -custom, indicating a TextChoice with a possible custom value
+            if key_parts[-1].endswith("-custom"):
```

Comment:
```suggestion
            elif key_parts[-1].endswith("-custom"):
```

Reply:
I don't believe there was any possible form submission which might trigger both, but your suggestion is logically correct.